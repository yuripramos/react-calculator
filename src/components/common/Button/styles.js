import React from "react";
import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { Link as RouterLink } from "react-router-dom";
import { white, green, green20, black30 } from "../../../styles/settings";

export const ButtonTag = styled.button`
  font-family: "RobotoSlab Regular";
  border-radius: ${rem(4)};
  border: 0;
  cursor: pointer;
  display: inline-block;
  font-size: ${rem(12)};
  letter-spacing: ${rem(0.5)};
  line-height: ${rem(45)};
  outline: none;
  padding: ${rem(5)};
  position: relative;
  text-align: center;
  text-decoration: none;
  color: ${white};
  transition: 0.2s ease-in-out color, 0.2s ease-in-out background,
    0.2s ease-in-out box-shadow, 0.2s ease-in-out border, 0.1s transform;
  white-space: nowrap;
  width: ${props => (props.width ? rem(props.width) : "100%")};
  user-select: none;
  opacity: ${props => (props.opacity && props.disabled ? props.opacity : 1)};
  background: transparent;
  transform: translateY(15px);

  ${media.md(css`
    width: ${props => (props.width ? rem(props.width) : "auto")};
  `)};

  ${media.lg(css`
    font-size: ${rem(13)};
    line-height: ${rem(48)};
  `)};

  :active {
    transform: translateY(${rem(1)});
  }
  :hover {
    background: ${white};
    color: ${black30};
    transform: translateY(20px);
  }

  ${({ isCallToAction }) =>
    isCallToAction &&
    css`
      background-color: ${green};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(100, 100, 100, 0.3);
      color: ${white};
      :hover {
        background: ${green20};
      }
    `};
`;

export const RouterLinkTag = ButtonTag.withComponent(
  ({ isCallToAction, ...props }) => <RouterLink {...props} /> // eslint-disable-line
);
