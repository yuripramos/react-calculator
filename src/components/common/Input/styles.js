import styled, { css } from "styled-components";
import { rem } from "../../../styles/tools";
import { white, grey30, black50 } from "../../../styles/settings";

export const InputField = styled.input`
  height: ${rem(45)};
  margin: 1em 0 4em;
  padding: 0;
  letter-spacing: 0.025em;
  background: transparent;
  text-transform: uppercase;
  color: white;
  border: none;
  border-bottom: solid thin white;
  color: #e8e8e8;
  color: rgba(255, 255, 255, 0.7);
  ::-webkit-input-placeholder {
    color: white;
    opacity: 0.35;
  }
  ::-moz-placeholder {
    color: white;
    opacity: 0.35;
  }
  ::-ms-input-placeholder {
    color: white;
    opacity: 0.35;
  }
  ::-moz-placeholder {
    color: white;
    opacity: 0.35;
  }
  :focus {
    outline: none;
    color: white;
    ::-webkit-input-placeholder {
      color: white;
      opacity: 0.7;
    }
  }
  ${({ width }) =>
    width &&
    css`
      width: ${width}%;
    `};
`;

export const InputWrapper = styled.div`
  width: 100%;
  display: inline-flex;
  position: relative;
  height: ${rem(26)};
  justify-content: center;
`;
