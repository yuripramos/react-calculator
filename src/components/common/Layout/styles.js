import styled, { css } from "styled-components";
import { media, rem } from "../../../styles/tools";

export const Content = styled.div`
  ${media.sm(css`
    min-height: calc(100vh - ${rem(250)});
  `)};
  ${media.md(css`
    min-height: calc(100vh - ${rem(250)});
  `)};
  ${media.lg(css`
    min-height: calc(100vh - ${rem(320)});
  `)};
`;
