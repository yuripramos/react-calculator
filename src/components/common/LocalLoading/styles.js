import styled, { keyframes } from "styled-components";
import { darkBlue } from "../../../styles/settings";
import { rem } from "../../../styles/tools";

const loading = keyframes`
  0% {
    transform: scale(1);
  }
  20% {
    transform: scale(1, 2.2);
  }
  40% {
    transform: scale(1);
  }
`;

export const Wrapper = styled.div`
  margin-top: ${rem(40)};
  position: relative;
  height: ${rem(40)};
`;

export const InnerWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
`;

export const Bar = styled.div`

  display: inline-block;
  width: ${rem(8)};
  height: ${rem(50)};
  margin: 0 ${rem(2)};
  border-radius: ${rem(5)};
  animation: ${loading} 1s ease-in-out infinite;
  background-color: ${darkBlue};

  :nth-child(1) {
    animation-delay: 0;
  }

  :nth-child(2) {
    animation-delay: 0.09s;
  }

  :nth-child(3) {
    animation-delay: 0.18s;
  }

  :nth-child(4) {
    animation-delay: 0.27s;
  }
`;
