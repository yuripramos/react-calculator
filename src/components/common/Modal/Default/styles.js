import styled from "styled-components";
import { rem } from "../../../../styles/tools";

export const DefaultContainer = styled.div`
  width: calc(100vw - ${rem(30)});
  max-width: ${rem(370)};
  padding-top: ${rem(16)};
`;
