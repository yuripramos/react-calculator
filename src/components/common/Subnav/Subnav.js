import React, { Component } from "react";
import { string, arrayOf, shape, object, func } from "prop-types";

import { Wrapper, TabContainer, Tab, Text, ContextTitle } from "./styles";

class Subnav extends Component {
  constructor(props) {
    super(props);

    this.state = {
      activeUrl: ""
    };
  }

  componentWillMount() {
    this.setState({ activeUrl: this.props.history.location.pathname });

    this.historyListenner = this.props.history.listen(location => {
      this.setState({ activeUrl: location.pathname });
    });
  }

  componentWillUnmount() {
    this.historyListenner();
  }

  onClickTab(url) {
    this.props.history.push(url);
  }

  render() {
    const { tabs, contextTitle } = this.props;
    const { activeUrl } = this.state;
    return (
      <Wrapper>
        {contextTitle && <ContextTitle>{contextTitle}</ContextTitle>}
        <TabContainer>
          {tabs.map((tab, index) => (
            <Tab
              data-test={`SubnavTab-${index}`}
              key={tab.url}
              onClick={() => this.onClickTab(tab.url)}
              active={activeUrl === tab.url}
            >
              <Text>{tab.text}</Text>
            </Tab>
          ))}
        </TabContainer>
      </Wrapper>
    );
  }
}

Subnav.defaultProps = {
  tabs: [],
  contextTitle: null
};

Subnav.propTypes = {
  contextTitle: string,
  tabs: arrayOf(
    shape({
      text: string,
      url: string
    })
  ),
  history: shape({
    location: object,
    listen: func
  }).isRequired
};

export default Subnav;
