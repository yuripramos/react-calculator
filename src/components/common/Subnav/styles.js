import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { white, grey30 } from "../../../styles/settings";

export const TabContainer = styled.ul`
  display: flex;
  position: relative;
  height: ${rem(25)};
  max-width: 100%;
  font-size: ${rem(12)};
`;

export const ContextTitle = styled.h1`
  font-family: "Roboto Bold", "Roboto";
  font-size: ${rem(20)};
  letter-spacing: ${rem(1.2)};
  color: ${white};
  text-align: center;
  white-space: nowrap;
  padding: ${rem(16)} 0;
  margin-top: -${rem(8)};
  line-height: 1.1;
  text-transform: capitalize;

  ${media.md(css`
    margin-top: 0;
    display: none;
    padding: ${rem(12)} 0;
  `)};
`;

export const Text = styled.span`
  font-family: "OpenSans Regular";
  font-size: ${rem(10)};
  letter-spacing: ${rem(1.3)};
  color: ${white};
  transition: 0.2s color;
  text-transform: uppercase;

  ${media.md(css`
    letter-spacing: ${rem(0.2)};
    font-size: ${rem(13)};
  `)};
`;

export const Tab = styled.li`
  align-items: flex-end;
  cursor: pointer;
  display: flex;
  width: ${rem(100)};
  height: ${rem(41)};
  justify-content: center;
  padding-bottom: ${rem(4)};
  position: relative;
  transition: 0.2s background;
  user-select: none;
  flex: 1;

  :after {
    background-color: ${white};
    bottom: -${rem(12)};
    content: "";
    height: ${rem(3)};
    left: 50%;
    opacity: 0;
    position: absolute;
    transform: translateX(-50%);
    transition: 0.3s;
    width: 0;
  }

  ${media.md(css`
    height: ${rem(10)};
    padding-bottom: ${rem(5)};

    :after {
      bottom: -${rem(9)};
    }
  `)};

  ${({ active }) =>
    active &&
    css`
      cursor: none;
      pointer-events: none;

      :after {
        width: 100%;
        opacity: 1;
      }

      ${Text} {
        color: ${white};
      }
    `};

  ${({ active }) =>
    !active &&
    css`
      :hover {
        ${Text} {
          color: ${white};
        }
      }
    `};
`;

export const Wrapper = styled.nav`
  ${media.md(css`
    background-color: ${grey30};
    height: ${rem(35)};
    display: flex;
    justify-content: center;
    padding-top: ${rem(7)};
  `)};
`;
