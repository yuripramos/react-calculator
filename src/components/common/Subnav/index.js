import { withRouter } from "react-router-dom";

import Subnav from "./Subnav";

export default withRouter(Subnav);
