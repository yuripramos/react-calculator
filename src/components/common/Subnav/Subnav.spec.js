import React from "react";
import { mount, shallow } from "enzyme";
import { Tab } from "./styles";

import Subnav from "./Subnav";

const tabs = [
  {
    text: "tab1",
    url: "/tab1"
  },
  {
    text: "tab2",
    url: "/tab2"
  }
];

let historyMock;

beforeEach(() => {
  historyMock = {
    location: {
      pathname: "/tab1"
    },
    listen: jest.fn(),
    push: jest.fn()
  };
});

describe("Subnav component", () => {
  it("should match snapshot with no tabs", () => {
    expect(shallow(<Subnav history={historyMock} />)).toMatchSnapshot();
  });

  it("should match snapshot with context title", () => {
    expect(
      shallow(<Subnav history={historyMock} contextTitle="context title" />)
    ).toMatchSnapshot();
  });

  it("should match snapshot with tabs", () => {
    expect(
      shallow(<Subnav history={historyMock} tabs={tabs} />)
    ).toMatchSnapshot();
  });

  it("should invoke listen", () => {
    shallow(<Subnav history={historyMock} tabs={tabs} />);

    expect(historyMock.listen).toHaveBeenCalledTimes(1);
  });

  it("should have a activeUrl state as same as location.pathname when component mounts", () => {
    const MountedSubnav = mount(<Subnav history={historyMock} tabs={tabs} />);

    expect(MountedSubnav.state("activeUrl")).toMatch(
      historyMock.location.pathname
    );
  });

  it("should call history.push with the same url as clicked tab", () => {
    const MountedSubnav = mount(<Subnav history={historyMock} tabs={tabs} />);

    MountedSubnav.find(Tab)
      .at(1)
      .simulate("click");

    expect(historyMock.push).toHaveBeenCalledWith(tabs[1].url);
  });
});
