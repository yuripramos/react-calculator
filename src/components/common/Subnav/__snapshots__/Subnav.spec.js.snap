// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subnav component should match snapshot with context title 1`] = `
<styled.nav>
  <styled.h1>
    context title
  </styled.h1>
  <styled.ul />
</styled.nav>
`;

exports[`Subnav component should match snapshot with no tabs 1`] = `
<styled.nav>
  <styled.ul />
</styled.nav>
`;

exports[`Subnav component should match snapshot with tabs 1`] = `
<styled.nav>
  <styled.ul>
    <styled.li
      active={true}
      data-test="SubnavTab-0"
      key="/tab1"
      onClick={[Function]}
    >
      <styled.span>
        tab1
      </styled.span>
    </styled.li>
    <styled.li
      active={false}
      data-test="SubnavTab-1"
      key="/tab2"
      onClick={[Function]}
    >
      <styled.span>
        tab2
      </styled.span>
    </styled.li>
  </styled.ul>
</styled.nav>
`;
